legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/NontemporalDeltaAbstract.java

```diff
@@ -78,6 +86,15 @@ public Void visitMaxVersionStrategy(MaxVersionStrategyAbstract maxVersionStrateg
                 {
                     throw new IllegalStateException("Cannot build NontemporalDelta, VersioningResolver is mandatory for MaxVersionStrategy");
                 }
+
+                MergeDataVersionResolver mergeStrategy = maxVersionStrategy.mergeDataVersionResolver().get();
+                if (mergeStrategy instanceof DigestBasedResolver)
```

Comment:
Can be moved to a common method as repeated in line 109-116

Reply:
sure, changed!